utils/linkage.py

```diff
@@ -44,6 +46,105 @@ def calculate_string_similarity(string1: str, string2: str) -> float:
     return float(td.jaro_winkler(string1.lower()[::-1], string2.lower()[::-1]))
 
 
+def calculate_row_similarity(
+    row1: pd.DataFrame, row2: pd.DataFrame, weights: np.array, comparison_func
+) -> float:
+    """Find weighted similarity of two rows in a dataframe
+
+    The length of the weights vector must be the same as
+    the number of selected columns.
+
+    This version is slow and not optimized, and will be
+    revised in order to make it more efficient. It
+    exists as to provide basic functionality. Once we have
+    the comparison function locked in, using .apply will
+    likely be easier and more efficient.
+
+    >>> d = {
```

Comment:
lets shift to testing this function per slack convo: https://uchicagodsi.slack.com/archives/C06ADK4FXNX/p1707149767586029